custom/aaa/ucr/reports/mpr_pop_totals_1_to_9.json

```diff
@@ -0,0 +1,225 @@
+{
+    "doc_type": "ReportConfiguration",
+    "domains": [
+        "icds-dashboard-qa",
+        "laurence-project-1",
+        "jessica-icds-cas",
+        "marissa-test",
+        "derek-icds-sandbox",
+        "priyanka-app",
+        "shrena-dev",
+        "aparatest",
+        "reach-sandbox",
+        "reach-dashboard-qa",
+        "reach-test",
+        "icds-test",
+        "icds-sql",
+        "icds-cas",
```

Comment:
Before I review this in depth, how critical is it that this needs to be added to icds-cas? I'd like to come up with a way forward that does not involve more reports deployed to TCL than is required for CAS

Reply:
The linked app is the main motivator for this to be in icds-cas. The end state of this ticket is that the relevant report modules are built in the master AWW app but hidden from CAS workers. My understanding is that for the linked app to pull successfully, it needs access to all static UCRs that are referenced in the app. If I'm mistaken in that (or if there's some way around it), I'm happy to drop it from icds-cas.

Reply:
It is accurate that domains do need access to that in order to link apps. However we cannot release this data source and report as is to TCL as it would have a large cost to add it there. Rough calculation right now are 13 columns (10 + implicit doc_id, inserted_at, repeat_iteration) * 8 bytes (undershooting) * 1 billion beneficiaries * 3 rows / beneficiary = ~300 GB raw data, and there'd be dead tuples as it gets updated and vacuumed and we'd need to index village owner id as we would not want sequential scans. We'd also need to add the supervisor id for the case to distribute it in the new citus database.

It would also make the restore longer because even if the report is hidden, it would still be doing the query for every restore.

In an ideal world where REACH was still being resourced, I'd probably add the village owner id to the current person cases v2 data source on icds and write this report using custom SQL to ensure it was ready for a TCL release, but I don't think that time will ever come.

I think this is fine for a pilot or demo scenario, but can't be deployed to TCL. So I think that I would be ok if we removed the icds-cas environment here so you could make the change on the dev stream, pull it to whatever reach app wants to have this report and then remove it from the dev stream, so that it would never be deployed to a TCL app. (personally I think that's a lot of work for this, but I'm not sure how important this is, its the best I can come up with at this time.)

Reply:
That all makes sense to me.

Before removing the report from icds-cas entirely, is there any way to have the report exist on the TCL server, but not actually query any data? Some sort of server-specific filter we can add onto it? I'm imagining a world where the report exists with an ID that could be referenced by the app, but the underlying data source always has 0 rows. Not sure if this is possible without development work (which would almost certainly not be worth it), but wanted to explore all possibilities before trying to juggle a crazy deploy process.

Reply:
I think that, technically as long as we don't add it to https://github.com/dimagi/commcare-cloud/blob/master/environments/icds-cas/public.yml#L223 then changes won't processed, but the app would still try to query the table which would still add some overhead to each request. It would also add app overhead since it would add report modules that aren't being used.

Reply:
OK, Wouter and I discussed it and think we can move forward with only deploying these to domains on the india server. I've cut the list down to only four domains: icds-cas (the master domain), reach-sandbox (for developer testing), reach-test (ironically for live deployment), and reach-dashboard-qa (for QA). Other alternatives are opening a can of worms that we'd rather not deal with.